canonical-web-and-design/maas-ui
# feat(ui): build device configuration form

## Done

- Built device configuration form
- Made `TagLinks` component work for both devices and machines

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/devices and click on a device
- Click on the "Configuration" tab
- Check that the device details are correct
- Click "Edit", change some values and submit the form
- Check that the device details successfully update
- Check that clicking on the tag links takes you to the device list with the correct filter applied

## Fixes

Fixes canonical-web-and-design/app-tribe#543

## Screenshot
![Peek 2021-12-03 13-06](https://user-images.githubusercontent.com/25733845/144541487-5103621a-3223-4205-873b-8dc3dd91e3e8.gif)
